src/audio_buffer.rs

```diff
@@ -0,0 +1,482 @@
+use std::rc::Rc;
+use std::sync::Arc;
+
+use crate::SampleRate;
+
+// @todo - put it there, but should leave in `BaseAudioContext`
+// let's just go with Wav files for now
+//
+// @note - what about using https://github.com/pdeljanov/Symphonia ?
```

Comment:
I really like this suggestion, we should definitely check out this crate and use it

Reply:
good